### ISSUE TITLE
Helm upgrade path from Opendistro-for-Elasticsearch 1.13

### ISSUE THREAD

**Issue by avthart** [opensearch-project/opensearch-devops#21]
We have a couple of questions regarding the new Opensearch project and 1.0.0 release.

We are currently using Opendistro-for-Elasticearch 1.13.x.

- What are the plans with the Helm chart for opendistro-for-es? Will this Helm chart be (re)used for Opensearch?
- When can we expect a release for Opensearch 1.0.0?

Also see https://github.com/opendistro-for-elasticsearch/opendistro-build/issues/780

Thanks.

**Comment by peterzhuamazon** [comment]
Hi @avthart we already have a helm OpenSearch chart in the repo https://github.com/opensearch-project/opensearch-devops/tree/main/Helm

The Dashboards one is pending on review: #10 

As for the upgrading path I am not sure whether the one we write for TAR works as well: https://opensearch.org/docs/upgrade-to/index/ since Helm uses docker.

I think @DandyDeveloper @dblock @TheAlgo can help on more on that.

Thanks.

**Comment by DandyDeveloper** [comment]
@avthart Chart wise, they are incompatible but can be structured in the same way as before effectively creating the same resources with the same supporting configs/secrets.

That being said, I haven't thoroughly tested my own current production Opendistro ES to check where pain points are. 

My understanding is for upgrading the version of ES, if you do a standard rollout via k8s for example, the Opensearch image can replace the OpenDistro ES image as if you were upgrading that way. 

However, chart wise, there's a big difference and compatibility was overall ignored in the initial creation of the chart. 

This is something I'll likely be looking at in the next month or so, as I'll be needing to migrate my production cluster :)

**Comment by TheAlgo** [comment]
> However, chart wise, there's a big difference and compatibility was overall ignored in the initial creation of the chart.
@DandyDeveloper 
While creating the initial version of the chart the following considerations were made:
1. Move away from monolithic chart. Decouple OpenSearch Dashboards from the OpenSearch chart.
2.  Give maximum customisability to the end users by making a generic chart. This way we can accommodate deployment strategies for all.

@avthart We can definitely have one chart which satisfies the upgrade from the `opendistro` chart in case we feel it is the best way to migrate to `OpenSearch`.

**Comment by DandyDeveloper** [comment]
@TheAlgo You may find we already can support an upgrade path since we're using the same resource types. The only big problem is it'll almost definitely result in downtime because the statefulset manifest is significantly different. 

I think it's hard to make a one size fits all for upgrade paths. In my case, I have a ton of special cases that never made it into the official Opendistro chart. 

I think the important thing to test is basic upgradeability, which needs testing. I may get time for this soonish.

**Comment by TheAlgo** [comment]
> @TheAlgo You may find we already can support an upgrade path since we're using the same resource types. The only big problem is it'll almost definitely result in downtime because the statefulset manifest is significantly different.
> 
> I think it's hard to make a one size fits all for upgrade paths. In my case, I have a ton of special cases that never made it into the official Opendistro chart.
> 
> I think the important thing to test is basic upgradeability, which needs testing. I may get time for this soonish.

I agree on it we can support the  upgrade path. I would love to understand about those special cases (I am curious) 😸
